### DISCUSSION TITLE
The engine "node" is incompatible with this module

### DISCUSSION THREAD

**Discussion by hardikbaldania** [community/community#130779] [Actions]
### Select Topic Area

Show & Tell

### Body

Hello,

I recently came across some issues with my CI/CD pipeline which was developed by my previous colleague. 

Idk what is the issue here since I'm new to this setup. I had previous auth issue with the github actions some how manage to fix it but took me like a week.

Requesting your assistance to this issue. I tried all the possible fixes and resolution but none worked for me like, updated the node.js version in my workflow ci.yml file. Added few steps related to clearing the cache for npm and checking the node version down the process. 

Here is the error: 
Installing plugin @salesforce/sfdx-scanner... ⣽ [2/4] Fetching packages...
error @typescript-eslint/parser@7.14.1: The engine "node" is incompatible with this module. Expected version "^18.18.0 || >=20.0.0". Got "18.15.0"
error Found incompatible module.
Installing plugin @salesforce/sfdx-scanner... failed
    Error: yarn add @salesforce/sfdx-scanner@latest --non-interactive 
    --mutex=file:/home/runner/.local/share/sfdx/yarn.lock 
    --preferred-cache-folder=/home/runner/.cache/sfdx/yarn --check-files 
    exited with code 1
Error: Process completed with exit code 1.

Please help with the fixes as I have deployment next week and this issues would probably get the deployment delayed.

Looking forward to hearing from you all community folks soon.

Thank you in advance and appreciate your time invested.

Regards,
Hardik

**Comment by joshjohanning** [comment]
Add (or tweak) the [`actions/setup-node`](https://github.com/actions/setup-node) action

```yml
- uses: actions/setup-node@v4
  with:
    node-version: 18 # or use 20
```

**Comment by hardikbaldania** [comment]
I already have done that adding the setup-node action, please have a look
jobs:
  validationqa:
    runs-on: ubuntu-latest
    if: ${{ github.actor != 'dependabot[bot]' }}
    steps:
      - uses: actions/checkout@v4
      - name: Use Node.js
        uses: actions/setup-node@v4
        with:
          node-version: '20.0.0'
      - name: Clear npm cache
        run: npm cache clean --force
      - name: Verify Node.js Version
        run: node -v
      - name: Install Salesforce CLI
        run: >

Yet it doesn't work and I get the same error with the node version: 18.15.0

**Reply by joshjohanning** [reply]
Is it possible the node condition is incorrect? Try just `18` to see what happens. Run the setup-node action before you need to reference it, of course.

**Comment by hardikbaldania** [comment]
I found the solution and was able to fix it myself. Thank you all for your efforts 👍

**Reply by agam-sharma** [reply]
What was the solution

**Reply by lianmerkdev** [reply]
see below

**Comment by lianmerkdev** [comment]
This happens as a result of using the old version of SF CLI. See [here](https://www.npmjs.com/package/sfdx-cli) SFDX CLI is deprecated. 

In your github action us SF CLI instead of SFDX CLI
so for example instead of https://developer.salesforce.com/media/salesforce-cli/sfdx/channels/stable/sf-linux-x64.tar.xz
use https://developer.salesforce.com/media/salesforce-cli/sf/channels/stable/sf-linux-x64.tar.xz

in this case the url is nearly the same, it's just the sfdx part that changes to sf
